src/App.jsx

```diff
@@ -14,45 +16,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_LENGTH = 'sortByLength';
+const SORT_ALPHABET = 'SortAlphabetically';
+
+function sortBy(goods, sortField, rev) {
+  const prepearedGoods = [...goods];
+
+  if (sortField === 'sortByLength' && rev % 2 !== 0) {
+    prepearedGoods.sort((curent, next) => curent.length - next.length);
+  }
+
+  if (sortField === 'sortByLength' && rev % 2 === 0) {
+    prepearedGoods.sort((curent, next) => next.length - curent.length);
+  }
+
+  if (sortField === 'SortAlphabetically' && rev % 2 !== 0) {
+    prepearedGoods.sort((curent, next) => curent.localeCompare(next));
+  }
+
+  if (sortField === 'SortAlphabetically' && rev % 2 === 0) {
+    prepearedGoods.sort((curent, next) => next.localeCompare(curent));
+  }
+
+  if (sortField === '' && rev % 2 === 0) {
+    prepearedGoods.reverse();
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return prepearedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [rev, checkRev] = useState(1);
```

Comment:
It's better to make isReversed boolean variable and change it to true on Reverse button click